cmd/stack.go

```diff
@@ -307,10 +307,13 @@ var stackUpdateCmd = &cobra.Command{
 
 		// Step 5b. Communicate with server to share progress of ...
 		if isNonInteractive() {
+			providerErrorDetected := false
+
 			fmt.Printf("Deploying %s stack with provider %s\n", stackConfig.Name, stackConfig.Provider)
 			go func() {
 				for update := range errorChan {
 					fmt.Printf("Error: %s\n", update)
+					providerErrorDetected = true
```

Comment:
This looks good for now, because the error handling is in a go-routine there is a potential race here if the eventChan closes before any errors are received, which is unlikely to be a problem. We can fix this later by merging these channels into a single event stream or by using a select statement to make sure they're all recieved in the same routine.